clang/lib/Driver/ToolChains/Arch/RISCV.cpp

```diff
@@ -166,8 +166,16 @@ void riscv::getRISCVTargetFeatures(const Driver &D, const llvm::Triple &Triple,
   if (Args.hasArg(options::OPT_ffixed_x31))
     Features.push_back("+reserve-x31");
 
+  bool IsCheri = llvm::find(Features, "+xcheri") != Features.end();
```

Comment:
Hm I thought I refactored this to use RISC-V ISAInfo?

Reply:
Yeah, you did, I just hadn't rebased (see the conflicts shown at the bottom of the page). Just testing the rebased branch now (which I also tweaked to stop re-parsing the features).